### PR TITLE
Checks input of factorial

### PR DESCRIPTION
**Pull request by dylanhernandez1**
This change simply just adds a feature to the factorial part to make sure that the input is valid. Specifically, it makes sure that the "n" variable input is valid by checking if it is greater than or equal to zero and if it is indeed a whole number. If it is not, the program prints that "n" is an invalid input. I agree to all of the terms below.

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?
* [x] Add/Update/Fix test cases.

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Dart files are placed inside an existing directory.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/factorial.dart [hunk 1]
```diff
@@ -1,7 +1,11 @@
 void main() {
   var n = 5;
-  var fac = factorial(n);
+  if(n>=0 && n.runtimeType == int){
```
Comment (akashgk) [inline maths/factorial.dart:3]: Instead of checking types, Why not replace `var n =5;` with `int n=5;`
This will help us avoid the unnecessary type checking.

File: maths/factorial.dart [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+    var fac = factorial(n);
   print("$n! = $fac"); /* output: 5! = 120 */
+  }else{
+    print("Invalid input. Input has to be a positive whole integer");
+  }
```
Comment (akashgk) [inline maths/factorial.dart:8]: can you add more tests to this file. Checkout other files to see how tests are written there.
[Two sum File](https://github.com/TheAlgorithms/Dart/blob/master/array/two_sum.dart)

Comment (akashgk) [reply maths/factorial.dart:8]: Let's remove these print statements and have proper tests for these functions. im in the process of upgrading the dart version and adding tests to old files so we can migrate to null safety and know all algorithms are working fine with Null safety.

**Review by akashgk — CHANGES_REQUESTED** [review]

File: maths/factorial.dart (hunk 1 above)
Comment (dylanhernandez1) [reply maths/factorial.dart:3]: We can do that, I would just say to leave it a variable in case if user input is needed. Whatever you want though.

File: maths/factorial.dart (hunk 2 above)
Comment (dylanhernandez1) [reply maths/factorial.dart:8]: You want me to put more tests in there for the factorials? Are they the same as the ones that you put on the [two sum file](https://github.com/TheAlgorithms/Dart/pull/220#:~:text=are%20written%20there.-,Two%20sum%20File,-Member)?

File: maths/factorial.dart (hunk 1 above)
Comment (akashgk) [reply maths/factorial.dart:3]: Since we are not taking any user input and dart has a great types. let's exploit that. 

you can add test for the same.

File: maths/factorial.dart (hunk 2 above)
Comment (dylanhernandez1) [reply maths/factorial.dart:8]: I can't think of any tests to add, sorry...let me know if there are specific tests I can add
